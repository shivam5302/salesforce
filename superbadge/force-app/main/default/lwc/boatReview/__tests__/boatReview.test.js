import { createElement } from 'lwc';
import BoatReview from 'c/boatReview';
import getAllReviews from '@salesforce/apex/BoatDataService.getAllReviews';
import mockGetRecord from './data/getReviews.json';

jest.mock(
	'@salesforce/apex/BoatDataService.getAllReviews',
	() => {
		const { createApexTestWireAdapter } = require('@salesforce/sfdx-lwc-jest');
		return {
			default: createApexTestWireAdapter(jest.fn())
		};
	},
	{ virtual: true }
);

jest.mock('c/fiveStarRating')


const flushPromises = async () => new Promise((resolve) => setTimeout(resolve, 0));

describe('c-boat-review', () => {
	afterEach(() => {
		// The jsdom instance is shared across test cases in a single file so reset the DOM
		while (document.body.firstChild) {
			document.body.removeChild(document.body.firstChild);
		}
	});

	it('TODO: test case generated by CLI command, please fill in test logic', () => {
		// Arrange
		const element = createElement('c-boat-review', {
			is: BoatReview
		});

		// Act
		document.body.appendChild(element);

		// Assert
		// const div = element.shadowRoot.querySelector('div');
		expect(1).toBe(1);
	});

	it('Wire Should show boat reviews', async () => {
			// Given
			const element = createElement('c-boat-review', {
				is: BoatReview
			});
			document.body.appendChild(element);
			
			getAllReviews.emit(mockGetRecord);
			// Wait for promises to resolve
			await flushPromises();

	
			// Then
			expect(element).toMatchSnapshot();
		});
});