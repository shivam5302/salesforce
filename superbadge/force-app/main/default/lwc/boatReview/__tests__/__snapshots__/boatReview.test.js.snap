// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-boat-review Wire Should show boat reviews 1`] = `
<c-boat-review
  __lwc_scope_token__=""
>
  #shadow-root(open)
    <!---->
    <div
      __lwc_scope_token__=""
      class="slds-feed, reviews-style, slds-is-relative, slds-scrollable_y"
    >
      <ul
        __lwc_scope_token__=""
        class="slds-feed__list"
      >
        <li
          __lwc_scope_token__=""
          class="slds-feed__item slds-m-around_small"
        >
          <article
            __lwc_scope_token__=""
            class="slds-post"
          >
            <header
              __lwc_scope_token__=""
              class="slds-post__header slds-media"
            >
              <div
                __lwc_scope_token__=""
                class="slds-media__figure"
              >
                <lightning-avatar
                  __lwc_scope_token__=""
                >
                  #shadow-root(open)
                </lightning-avatar>
              </div>
              <div
                __lwc_scope_token__=""
                class="slds-media__body"
              >
                <div
                  __lwc_scope_token__=""
                  class="slds-grid slds-grid_align-spread slds-has-flexi-truncate"
                >
                  <p
                    __lwc_scope_token__=""
                  >
                    <span
                      __lwc_scope_token__=""
                    >
                      xyz
                    </span>
                    --
                    <span
                      __lwc_scope_token__=""
                    >
                      companyName
                    </span>
                  </p>
                </div>
                <p
                  __lwc_scope_token__=""
                  class="slds-text-body_small"
                >
                  <lightning-formatted-date-time
                    __lwc_scope_token__=""
                  >
                    #shadow-root(open)
                  </lightning-formatted-date-time>
                </p>
              </div>
            </header>
            <div
              __lwc_scope_token__=""
              class="slds-text-longform"
            >
              <p
                __lwc_scope_token__=""
                class="slds-text-title_caps"
              >
                boatName
              </p>
              commoneeeeeeeeeent
            </div>
            <c-five-star-rating
              __lwc_scope_token__=""
            >
              #shadow-root(open)
            </c-five-star-rating>
          </article>
        </li>
      </ul>
    </div>
    <!---->
</c-boat-review>
`;
